pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractReplicator.java

```diff
@@ -64,10 +70,35 @@ public abstract class AbstractReplicator {
 
     protected static final AtomicReferenceFieldUpdater<AbstractReplicator, State> STATE_UPDATER =
             AtomicReferenceFieldUpdater.newUpdater(AbstractReplicator.class, State.class, "state");
-    private volatile State state = State.Stopped;
-
-    protected enum State {
-        Stopped, Starting, Started, Stopping
+    @VisibleForTesting
+    @Getter
+    protected volatile State state = State.Stopped;
+
+    public enum State {
+        /**
+         * This enum has two mean meanings：Init, Stopped.
+         * Regarding the meaning "Stopped", only {@link PersistentTopic#checkGC} will call {@link #disconnect},
+         *   so this method only be used by {@link PersistentTopic#checkGC} now.
+         * TODO After improving the method {@link #disconnect)}, we should rename "Stopped" to "Init".
+         */
+        // The internal producer is stopped.
+        Stopped,
+        // Trying to create a new internal producer.
+        Starting,
+        // The internal producer has started, and tries copy data.
+        Started,
+        /**
+         * @Deprecated Only {@link PersistentTopic#checkGC} will call {@link #disconnect}, so this method only be
+         *  used by {@link PersistentTopic#checkGC} now.
+         * TODO After improving the method {@link #disconnect)}, this enum should be removed.
+         */
+        @Deprecated
+        // The internal producer is trying to stop.
+        Stopping,
```

Comment:
We still use `Stopping` when disconnecting the replicator. Why do we need add `@Deprecated` annotation?

BTW, is it better to change `Stopping`, `Stopped` to `Disconnecting`, `Disconnected`? Honestly, It's not easy to understand what is the difference between stop, terminate. But disconnecting will be more better in this case.

Reply:
Renamed